app/HutchManager/Services/QueryTranslator.cs

```diff
@@ -0,0 +1,73 @@
+using HutchManager.Data.Entities;
+using HutchManager.Dto;
+
+namespace HutchManager.Services;
+
+public class QueryTranslator
+{
+  public interface IQueryTranslator<T>
+  { 
+    ROCratesQuery Translate(T input);
+  }
+  public class RquestQueryTranslator: IQueryTranslator<RquestQueryTask>{
+    public ROCratesQuery Translate(RquestQueryTask job)
+    {
+      
+      ROCratesQuery roCratesQuery = new();
+      var Graph = new ROCratesQuery().Graphs;
+      //Add ActivitySourceID
+      Graph.Add(new ROCratesQuery.ROCratesGraph());
+      Graph[0].Context = "https://schema.org";
+      Graph[0].Type = "PropertyValue";
+      Graph[0].Name = "activity_source_id";
+      Graph[0].Value = job.ActivitySourceId.ToString();
+      
+      // Add Job ID
+      Graph.Add(new ROCratesQuery.ROCratesGraph());
+      Graph[1].Context = "https://schema.org";
+      Graph[1].Type = "PropertyValue";
+      Graph[1].Name = "activity_source_id";
+      Graph[1].Value = job.JobId;
+      
+      var item = new List<ROCratesQuery.Item>();
+      
+      roCratesQuery.Context = "https://w3id.org/ro/crate/1.1/context";
+      int length = job.Query.Groups.Count()+2;
+      //For each group in an rquestQuery create a Graph in roCratesQuery
+      for (int i=2;i<length;i++)
```

Comment:
I think you can eliminate `i` and `j` here (and therefore the starting at 2 / going to +2 loop, which isn't instantly grokable) by switching to object initialization as well.

Also if you reverse the order of initialisation (inner items first) you don't need to keep as many references to stuff around, as you can immediately assign the inner values to the outer properties on construction

e.g.

```csharp
foreach(var group in job.Query.Groups)
{
  var graph = new Graph()
  {
    // group properties
    // ...
    NumberOfItems = group.Rules.Count()
    ItemListElements = group.Rules.Select(rule =>
      // Select maps each item of a collection to a new item - it's like a functional `ForEach` provided you want to return a value into a list for each iteration. which we do - since we are building a list of Item; one for each Rule
      new Item
      {
        // item properties...
        AdditionalProperties:  new()
        {
          // properties properties
        }
      })
  };
  graphs.Add(graph);
}
```

Obviously you don't _have_ to inline stuff quite this far, using `Select()` etc. But I would definitely advocate a `ForEach()` style over keeping and reusing an index (`i`, `j` etc), if you don't need to.